.github/workflows/dotnet-upgrade.yml

```diff
@@ -0,0 +1,228 @@
+        shell: bash
+        run: |
+          current_patch_version=$(jq .sdk.version ./src/global.json | xargs)
+          current_major_minor_version=$(cut -d '.' -f 1,2 <<< "$current_patch_version")
+
+          echo "DOTNET_CURRENT_VERSION=${current_patch_version}" >> $GITHUB_OUTPUT
+          echo "DOTNET_MINOR_VERSION=${current_major_minor_version}" >> $GITHUB_OUTPUT
+      - name: Check patch version
+        id: fetch_latest_patch
+        shell: bash
+        run: |
+          latest_patch_version=$(curl -sb -H "Accept: application/json" "https://dotnetcli.blob.core.windows.net/dotnet/Sdk/${{ steps.fetch_current_minor_version.outputs.DOTNET_MINOR_VERSION }}/latest.version")
+          current_patch_version=$(jq .sdk.version ./src/global.json | xargs)
+
+          should_update=0
+          [ $current_patch_version != $latest_patch_version ] && should_update=1
+
+          # check if git branch already exists for the upgrade
+          branch_already_exists=0
+          
+          if git ls-remote --heads --exit-code origin refs/heads/feature/dotnetsdk-upgrade/${latest_patch_version};
+          then 
+            branch_already_exists=1
+            should_update=0
+          fi
+          echo "DOTNET_LATEST_VERSION=${latest_patch_version}" >> $GITHUB_OUTPUT
+          echo "SHOULD_UPDATE=${should_update}" >> $GITHUB_OUTPUT
+          echo "BRANCH_EXISTS=${branch_already_exists}" >> $GITHUB_OUTPUT
+      - name: Create annotation for branch error
+        if: ${{ steps.fetch_latest_patch.outputs.BRANCH_EXISTS == 1 }}
+        run: echo "::error ::Branch feature/dotnetsdk-upgrade/${{ steps.fetch_latest_patch.outputs.DOTNET_LATEST_VERSION }} already exists. Please address take a look and delete that branch if you wish to recreate"
+      - name: Create annotation for build error
+        if: ${{ steps.fetch_latest_patch.outputs.SHOULD_UPDATE == 0 && steps.fetch_latest_patch.outputs.BRANCH_EXISTS == 0 }}
+        run: echo "::warning ::Latest DotNet SDK patch is ${{ steps.fetch_latest_patch.outputs.DOTNET_LATEST_VERSION }}, and we are on ${{ steps.fetch_latest_patch.outputs.DOTNET_CURRENT_VERSION }}. No need to update"
+      - name: Update patch version
+        if: ${{ steps.fetch_latest_patch.outputs.SHOULD_UPDATE == 1 && steps.fetch_latest_patch.outputs.BRANCH_EXISTS == 0 }}
+        shell: bash
+        run: |
+          patch_version="${{ steps.fetch_latest_patch.outputs.DOTNET_LATEST_VERSION }}"
+          current_version="${{ steps.fetch_current_minor_version.outputs.DOTNET_CURRENT_VERSION }}"
+
+          # Update globals
+          echo $(cat ./src/global.json | jq --arg patch_version "$patch_version" '.sdk.version = $patch_version') > ./src/global.json
+
+          # Update devcontainer
+          echo $(cat ./.devcontainer/devcontainer.json | jq --arg patch_version "$patch_version" '.features."ghcr.io/devcontainers/features/dotnet".version = $patch_version') > ./.devcontainer/devcontainer.json
```

Comment:
can we make sure the output json is pretty printed with indent?
![image](https://github.com/actions/runner/assets/1750815/6ecc34f6-a40c-4ea4-a494-4ce9fac072f9)

Reply:
Ahh whoops, that was happening due to the echo output being piped into the file. Fixed in latest!